### ISSUE TITLE
Unable to send email

### ISSUE THREAD

**Issue by MineshK** [gophish/gophish#1464]
What version of Gophish are you using?: 
I have two instances of GoPhish as below:

Docker build by matteoggl - 0.6.0
Built from source on CentOS on GCP - 0.7.1

Brief description of the issue: 

When I test email sending or real campaign sending from the Docker instance, email works fine. 

I am now trying to get it working on the GCP instance, but sending from the same account fails, with the error "Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout" (x.x.x.x = IP address of my email server).

I have set up ingress and egress rules for port 25 within GCP, but still problematic.

My cPanel server says I can use the following ports:

SSL:
IMAP Port: 993 POP3 Port: 995
SMTP Port: 465

Non-SSL:
IMAP Port: 143 POP3 Port: 110
SMTP Port: 26

What are you expecting to see happen? :

I would expect the email to send, as it does within the Docker instance.

What are you seeing happen?

Error: 
Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout

Please provide any terminal output that may be relevant below:

```
time="2019-05-21T21:11:49Z" level=warning msg="Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout"
time="2019-05-21T21:11:49Z" level=info msg="y.y.y.y - - [21/May/2019:21:10:09 +0000] \"POST /api/util/send_test_email HTTP/2.0\" 500 127 \"https://z.z.z.z:3333/sending_profiles\" \"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.157 Safari/537.36\""
time="2019-05-21T21:11:49Z" level=error msg="Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout"
```
Note: 
x.x.x.x = Sending email server IP
y.y.y.y = IP I am connecting to HTTP Server from
z.z.z.z = GCP Server IP Address

I would be grateful for some help - I can't control the firewall on the email server, so unable to open port 25 on that - though I am unsure if that is the issue, as it worked from the Docker version.

Thanks

**Comment by S0larflare** [comment]
You say that cPanel is telling you that you can use port 26. Is that a typo, as you’re using port 25?

Sent from my iPhone

On 21 May 2019, at 22:29, MineshK <notifications@github.com<mailto:notifications@github.com>> wrote:


What version of Gophish are you using?:
I have two instances of GoPhish as below:

Docker build by matteoggl - 0.6.0
Built from source on CentOS on GCP - 0.7.1

Brief description of the issue:

When I test email sending or real campaign sending from the Docker instance, email works fine.

I am now trying to get it working on the GCP instance, but sending from the same account fails, with the error "Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout" (x.x.x.x = IP address of my email server).

I have set up ingress and egress rules for port 25 within GCP, but still problematic.

My cPanel server says I can use the following ports:

SSL:
IMAP Port: 993 POP3 Port: 995
SMTP Port: 465

Non-SSL:
IMAP Port: 143 POP3 Port: 110
SMTP Port: 26

What are you expecting to see happen? :

I would expect the email to send, as it does within the Docker instance.

What are you seeing happen?

Error:
Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout

Please provide any terminal output that may be relevant below:

time="2019-05-21T21:11:49Z" level=warning msg="Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout"
time="2019-05-21T21:11:49Z" level=info msg="y.y.y.y - - [21/May/2019:21:10:09 +0000] "POST /api/util/send_test_email HTTP/2.0" 500 127 "https://z.z.z.z:3333/sending_profiles\<https://z.z.z.z:3333/sending_profiles%5C>" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.157 Safari/537.36""
time="2019-05-21T21:11:49Z" level=error msg="Max connection attempts exceeded - dial tcp x.x.x.x:25: i/o timeout"

Note:
x.x.x.x = Sending email server IP
y.y.y.y = IP I am connecting to HTTP Server from
z.z.z.z = GCP Server IP Address


I would be grateful for some help - I can't control the firewall on the email server, so unable to open port 25 on that - though I am unsure if that is the issue, as it worked from the Docker version.

Thanks

**Comment by razermouse** [comment]
First of all, GCP blocks all traffic on port 25 outbound no matter what rules you configure. To get email outbound working, you have to use Sendgrid, Mailgun, or something similar. Don't bother submitting a request for them to open it - they can't and won't. This is the reason why we moved over to AWS.

**Comment by MineshK** [comment]
Thanks - managed to get this sorted (should have gone to sleep and tried again with some sleep!).

I looked at smtp.go and realised the code uses port 25 only if one isn't given a port - which led me to try adding another port in my email settings - I didn't realise I could until this point.

I had suspected as such about GCP, so thank you for confirming - this explained why port 25 worked from my docker instance, but not from GCP.

Thanks for the help!
